x-pack/solutions/security/plugins/security_solution/public/cases/pages/index.tsx

```diff
@@ -41,24 +53,56 @@ const CaseContainerComponent: React.FC = () => {
 
   const interactionsUpsellingMessage = useUpsellingMessage('investigation_guide_interactions');
 
+  const AIForSOC = capabilities[SECURITY_FEATURE_ID].configurations;
+
   const showAlertDetails = useCallback(
     (alertId: string, index: string) => {
-      openFlyout({
-        right: {
-          id: DocumentDetailsRightPanelKey,
-          params: {
-            id: alertId,
-            indexName: index,
-            scopeId: TimelineId.casePage,
+      // TODO We shouldn't have to check capabilities here, this should be done at a much higher level.
+      //  https://github.com/elastic/kibana/issues/218741
+      //  For the AI for SOC we need to show the AI alert flyout.
+      if (AIForSOC) {
+        openFlyout({
+          right: {
+            id: IOCPanelKey,
+            params: {
+              id: alertId,
+              indexName: index,
+            },
+          },
+        });
+      } else {
+        openFlyout({
+          right: {
+            id: DocumentDetailsRightPanelKey,
+            params: {
+              id: alertId,
+              indexName: index,
+              scopeId: TimelineId.casePage,
+            },
           },
-        },
-      });
-      telemetry.reportEvent(DocumentEventTypes.DetailsFlyoutOpened, {
-        location: TimelineId.casePage,
-        panel: 'right',
-      });
+        });
+        telemetry.reportEvent(DocumentEventTypes.DetailsFlyoutOpened, {
+          location: TimelineId.casePage,
+          panel: 'right',
+        });
+      }
+    },
+    [capabilities, openFlyout, telemetry]
+  );
+
+  const renderAlertsTable = useCallback(
+    (props: CaseViewAlertsTableProps) => {
+      console.log('CaseContainerComponent props', props);
```

Comment:
leftover?

Reply:
Oops, indeed! Sorry about that, it's removed!